internal/pkg/runtime/engines/singularity/rpc/server/server_linux.go

```diff
@@ -33,6 +38,52 @@ func (t *Methods) Mount(arguments *args.MountArgs, reply *int) (err error) {
 	return err
 }
 
+// Crypt decrypts the loop device
+func (t *Methods) Decrypt(arguments *args.CryptArgs, reply *string) (err error) {
+
+	sylog.Debugf("In Crypt RPC")
+	sylog.Debugf("Crypt RPC running in PID %d", os.Getpid())
+	sylog.Debugf("Loop device is %s", arguments.Loopdev)
+
+	fmt.Print("Enter the password to decrypt File System: ")
+	password, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		sylog.Fatalf("Error parsing input: %s", err)
+	}
+	fmt.Println()
+
+	crypt_dev := &crypt.Device{
+		MaxDevices: 256,
+	}
+
+	cdev_str, err := crypt_dev.GetCryptDevice()
+
+	sylog.Debugf("Crypt device is %s\n", cdev_str)
+
+	cmd := exec.Command("/sbin/cryptsetup", "luksOpen", arguments.Loopdev, cdev_str, "-v", "--debug")
```

Comment:
let's not depend on PATH lookups inside privileged code.

If you need to resolve the path to cryptsetup, do it at build time, and introduce a configuration variable so that the admin that override that location.